platform-sdk/docs/proposals/TSS-Encryption-Key-Automation/TSS-Encryption-Key-Automation.md

```diff
@@ -0,0 +1,233 @@
+`shares of shares`. Aggregating the decrypted `shares of shares` allows the node to recover its private
+share keys.
+
+While the public key must be known by all nodes, the private `tssEncryptionKey` must not be known by any other node.
+The security of the ledger id and the ledger's private signing key is dependent on the private `tssEncryptionKey`
+remaining private to each node. If a single party were to obtain enough of the private `tssEncryptionKeys`, they
+would be able to recover the private shares of multiple nodes and aggregate a threshold number of private shares to
+recover the ledger's private signing key.
+
+The reason that the `tssEncryptionKey` must be maintained long-term is that the public key must be known by other
+nodes before they can generate private shares for the node. If the `tssEncryptionKey` were to change, the node would
+not be able to participate in TSS signing or re-keying until the next candidate roster is keyed and adopted.
+
+To minimize the number of node operator touch points, the life-cycle of the `tssEncryptionKey` will be automated.
+
+### Dependencies, Interactions, and Implications
+
+This proposal should be delivered with or before the delivery of `TSS-Ledger-ID`, which will attempt to key a candidate
+roster nightly. The `TSS-Ledger-ID` capability cannot be turned on until enough nodes know the public
+`tssEncryptionKey` of enough of the other nodes.
+
+### Requirements
+
+1. The public `tssEncryptionKey` of a node must be recorded in the state and publicly known by other nodes.
+2. The private `tssEncryptionKey` of a node must be kept private to that node.
+3. The lifecycle of the `tssEncryptionKey` should be automated.
+
+### Design Decisions
+
+* The private `tssEncryptionKey` will be stored in the same directory as the node's private gossip signing key.
+* The public `tssEncryptionKey` will be stored in the `TssBaseService` state.
+* A new `TssEncryptionKeyTransaction` will be gossiped, come to consensus, and handled by the `TssBaseService` to
+  update the public `tssEncryptionKey` in the state.
+  * This new transaction is system generated only and not user generated.  It should be rejected at the HAPI gateway
+    for user generated transactions.
+
+#### Alternatives Considered
+
+The alternative to the automated lifecycle is to have the node operators manually generate the `tssEncryptionKey`
+before the node starts and provide the public key to the network through a transaction signed by the node admin key.
+
+Requiring human intervention for this key increases friction that can easily be avoided. This saves all the testing
+labor related to testing human workflows.
+
+---
+
+## Changes
+
+The following changes to the system are required.
+
+### Architecture and/or Components
+
+The `TssBaseService` already exists with its own state.
+
+* This proposal adds a new `Map<Long, Bytes>` to the state for storing the public key of each node.
```

Comment:
Please update the doc to clarify if this is a virtual map, or a protobuf map.

If it is the latter, I recall that a decision was made to never use protobuf maps in the state because of their non-deterministic nature when it comes to serialization (which doesn't necessarily apply to PBJ in our own implementation, but may apply to 3rd-party implementations.)

If you still see this as a protobuf map, I'd like @jsync-swirlds to weigh in.

Reply:
A Virtual Map is good, but @anthony-swirldslabs is correct that we should not use protocol buffer maps in state due to the specified non-deterministic characteristic of protocol buffer map iteration.

Reply:
virtual map. will update.